src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -805,23 +822,29 @@ protected function {$methodName}($lazyInitialization)
         $inlinedDefinitions = $this->getDefinitionsFromArguments(array($definition));
         $constructorDefinitions = $this->getDefinitionsFromArguments(array($definition->getArguments(), $definition->getFactory()));
         $otherDefinitions = new \SplObjectStorage();
+        $serviceCalls = array();
 
         foreach ($inlinedDefinitions as $def) {
             if ($def === $definition || isset($constructorDefinitions[$def])) {
                 $constructorDefinitions[$def] = $inlinedDefinitions[$def];
             } else {
                 $otherDefinitions[$def] = $inlinedDefinitions[$def];
             }
+            $arguments = array($def->getArguments(), $def->getFactory(), $def->getProperties(), $def->getMethodCalls(), $def->getConfigurator());
+            $this->getServiceCallsFromArguments($arguments, $serviceCalls, false, $id);
```

Comment:
This factorizes computing `$serviceCalls` once instead of doing it in each method that requires the result.